### PR TITLE
Embedding third-party dependencies

### PR DESCRIPTION
**Pull request by testableapple**
### 🔗 Issue Link

Resolve https://github.com/GetStream/ios-issues-tracking/issues/601

### 🎯 Goal

Embed third-party dependencies

### 🧪 Testing

- [x] SPM integration
- [x] CocoaPods integration

### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Affected documentation updated (docusaurus, tutorial, CMS (task created)

### CODE REVIEW

File: Sources/StreamChatSwiftUI/ChatChannel/Gallery/ZoomableScrollView.swift [hunk 1]
```diff
@@ -89,14 +89,14 @@ private struct ZoomableScrollViewImpl<Content: View>: UIViewControllerRepresenta
...
-            doubleTapCancellable = doubleTap.sink { [unowned self] in handleDoubleTap() }
+                } as! any Cancellable // FIXME
+            doubleTapCancellable = doubleTap.sink { [unowned self] in handleDoubleTap() } as! any Cancellable // FIXME
         }
 
         func update(content: Content, doubleTap: AnyPublisher<Void, Never>) {
             coordinator.hostingController.rootView = content
             scrollView.setNeedsUpdateConstraints()
-            doubleTapCancellable = doubleTap.sink { [unowned self] in handleDoubleTap() }
+            doubleTapCancellable = doubleTap.sink { [unowned self] in handleDoubleTap() } as! any Cancellable // FIXME
```
Comment (testableapple) [inline Sources/StreamChatSwiftUI/ChatChannel/Gallery/ZoomableScrollView.swift:99]: @martinmitrevski, could you please take a look at what went wrong here?

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,10 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 # Upcoming
 
+### ⚠️ Important
+
+- Dependencies are no longer exposed (this includes Nuke and SwiftyGif). If you were using those dependencies we were exposing, you would need to import them manually.
```
Comment (martinmitrevski) [inline CHANGELOG.md:8]: Maybe also add that they should "Reset package caches" after integrating? I had issues until I did that.

File: Package.swift [hunk 3]
```diff
@@ -17,13 +17,12 @@ let package = Package(
...
     ],
     dependencies: [
-        .package(url: "https://github.com/GetStream/stream-chat-swift.git", from: "4.39.0"),        
-        .package(url: "https://github.com/kean/Nuke.git", .exact("11.3.1"))
+        .package(url: "https://github.com/GetStream/stream-chat-swift.git", from: "4.39.0"),
     ],
     targets: [
         .target(
             name: "StreamChatSwiftUI",
-            dependencies: [.product(name: "StreamChat", package: "stream-chat-swift"), "Nuke", .product(name: "NukeUI", package: "Nuke")],
```
Comment (martinmitrevski) [inline Package.swift:26]: I still see Difference, Swifter and the tools. Can we also get rid of those? (not in this PR)

**Review by martinmitrevski — APPROVED** [review]
LGTM ✅

File: Package.swift (hunk 3 above)
Comment (testableapple) [reply Package.swift:26]: Those come from StreamChat. They will be gone as soon as https://github.com/GetStream/stream-chat-swiftui/pull/376 is merged

File: CHANGELOG.md (hunk 2 above)
Comment (testableapple) [reply CHANGELOG.md]: Done 👍

**Pull request merged**